src/visualization/types.jl

```diff
@@ -348,58 +348,77 @@ end
+    y = reshape(view(cache.elements.node_coordinates, 2, :, :, :), n_nodes_2d, n_elements)
+    u_extracted = StructArray{SVector{nvars, uEltype}}(ntuple(_->similar(x, (n_nodes_2d, n_elements)), nvars))
+    for element in eachelement(dg, cache)
+      sk = 1
+      for j in eachnode(dg), i in eachnode(dg)
+        u_node = get_node_vars(u, equations, dg, i, j, element)
+        u_extracted[sk, element] = u_node
+        sk += 1
+      end
+    end
+    # interpolate to volume plotting points
+    xplot, yplot = plotting_interp_matrix*x, plotting_interp_matrix*y
+    uplot = StructArray{SVector{nvars, uEltype}}(map(x->plotting_interp_matrix*x,
+                                                   StructArrays.components(u_extracted)))
 
-  # construct a triangulation of the plotting nodes
-  t = reference_plotting_triangulation((r_plot, s_plot))
+    xfp, yfp, ufp = mesh_plotting_wireframe(u_extracted, mesh, equations, dg, cache; nvisnodes=nvisnodes)
 
-  # extract x,y coordinates and solutions on each element
-  uEltype = eltype(u)
-  nvars = nvariables(equations)
-  x = reshape(view(cache.elements.node_coordinates, 1, :, :, :), n_nodes_2d, n_elements)
-  y = reshape(view(cache.elements.node_coordinates, 2, :, :, :), n_nodes_2d, n_elements)
-  u_extracted = StructArray{SVector{nvars, uEltype}}(ntuple(_->similar(x, (n_nodes_2d, n_elements)), nvars))
-  for element in eachelement(dg, cache)
-    sk = 1
-    for j in eachnode(dg), i in eachnode(dg)
-      u_node = get_node_vars(u, equations, dg, i, j, element)
-      u_extracted[sk, element] = u_node
-      sk += 1
-    end
-  end
+    transform_to_solution_variables!(uplot, solution_variables_, equations)
+    transform_to_solution_variables!(ufp, solution_variables_, equations)
 
-  # interpolate to volume plotting points
-  xplot, yplot = plotting_interp_matrix*x, plotting_interp_matrix*y
-  uplot = StructArray{SVector{nvars, uEltype}}(map(x->plotting_interp_matrix*x,
-                                                   StructArrays.components(u_extracted)))
+    return PlotData2DTriangulated(xplot, yplot, uplot, t, xfp, yfp, ufp, variable_names)
 
-  xfp, yfp, ufp = mesh_plotting_wireframe(u_extracted, mesh, equations, dg, cache; nvisnodes=nvisnodes)
+  # In the 3D-case a 'PlotData2DCartesian' object is created.
+  elseif ndims(mesh) == 3
 
-  # convert variables based on solution_variables mapping
-  solution_variables_ = digest_solution_variables(equations, solution_variables)
-  variable_names = SVector(varnames(solution_variables_, equations))
+    # Set nvisnodes to 32 on default.
+    if nvisnodes == nothing
+      nvisnodes = 32
+    end
 
-  transform_to_solution_variables!(uplot, solution_variables_, equations)
-  transform_to_solution_variables!(ufp, solution_variables_, equations)
+    # Convert data.
+    nodes = cache.elements.node_coordinates
+    data_on_plane, plane = unstructured_3d_to_2d_plane(nodes, u; elevations, slice, point, nvisnodes)
```

Comment:
This works for both the `StructuredMesh` and the `P4estMesh` in 3D, correct? Do we have something similar for `DGMultiMesh`es in 3D?

Reply:
Yes it does. Unfortunately there is nothing similar for DGMultiMeshes in 3D yet.

Reply:
Okay, thanks for confirming. Is the issue documenting the progress up to date?

Reply:
@FelipeSantillan would you also mind adding the comment `TODO: DGMulti. Extend to DGMulti meshes` as a reminder?